src/kontrol/prove.py

```diff
@@ -207,18 +207,65 @@ def init_and_run_proof(test: FoundryTest) -> Proof:
                 use_gas=prove_options.use_gas,
             )
 
+            cut_point_rules = KEVMSemantics.cut_point_rules(
+                prove_options.break_on_jumpi,
+                prove_options.break_on_calls,
+                prove_options.break_on_storage,
+                prove_options.break_on_basic_blocks,
+            )
+            if prove_options.break_on_foundry:
+                cut_point_rules += [
```

Comment:
Adding only `FOUNDRY-CHEAT-CODES.foundry.call` should be enough, since it captures all the calls made to the cheat code address.

Reply:
We want to break on _every_ rule that is added in this file, not just on "is something happening for Foundry". This is to make sure we're maknig basic blocks that don't contain _any_ rules from this file in it.

Reply:
Can we have `FOUNDRY-CHEAT-CODES.foundry` instead? Are we supporting partial matching? All the rules enumerated here make this list easy to get out of date.

Reply:
I don't think partial matching is supported, but I guess that's something we could ask the backend team about.

I suppose we could have pyk instead read the names of the rules directly from the definition, let me see if that's possible.

Reply:
I've done this, now it's enumerated directly from the definition instead.